bridge/gitlab/import.go

```diff
@@ -59,42 +55,30 @@ func (gi *gitlabImporter) Init(_ context.Context, repo *cache.RepoCache, conf co
 // ImportAll iterate over all the configured repository issues (notes) and ensure the creation
 // of the missing issues / comments / label events / title changes ...
 func (gi *gitlabImporter) ImportAll(ctx context.Context, repo *cache.RepoCache, since time.Time) (<-chan core.ImportResult, error) {
-	gi.iterator = iterator.NewIterator(ctx, gi.client, 10, gi.conf[confKeyProjectID], since)
+
 	out := make(chan core.ImportResult)
 	gi.out = out
 
 	go func() {
-		defer close(gi.out)
+		defer close(out)
 
-		// Loop over all matching issues
-		for gi.iterator.NextIssue() {
-			issue := gi.iterator.IssueValue()
+		for issue := range Issues(ctx, gi.client, gi.conf[confKeyProjectID], since) {
 
-			// create issue
 			b, err := gi.ensureIssue(repo, issue)
 			if err != nil {
 				err := fmt.Errorf("issue creation: %v", err)
 				out <- core.NewImportError(err, "")
 				return
 			}
 
-			// Loop over all notes
-			for gi.iterator.NextNote() {
-				note := gi.iterator.NoteValue()
-				if err := gi.ensureNote(repo, b, note); err != nil {
-					err := fmt.Errorf("note creation: %v", err)
-					out <- core.NewImportError(err, entity.Id(strconv.Itoa(note.ID)))
-					return
+			for _, e := range SortedEvents(IssueEvents(ctx, gi.client, issue)) {
```

Comment:
If I understand that right, this force pulling all the events (in memory on top of that) to sort them, and only actually import after that. Is there not a way to pull them already ordered?
If the problem is merging the 3 sources of event together, this could be done "on the fly" by picking the earliest of the 3 head, assuming each of those source are ordered already.

Reply:
You are right! I would make git-bug feel snapier, I like it. However I am not sure when I am able to give it a try.

Reply:
I went ahead and implemented that. From a code perspective that works, but I have no idea it that actually works with gitlab. I guess the CI will tell.

Reply:
Ho well, the CI will not tell because this PR is in your repo. It has to be pushed in mine (which you will be able to do if you accept the invite) to have the proper tokens. In the mean time, I'll push that myself into a new PR.